src/core/patchers/BrowserWindowPatcher.ts

```diff
@@ -9,19 +9,27 @@ export type PatchFunction = (
 ) => object;
 export type UnpatchFunction = () => boolean;
 
+declare type KernelWebContents = electron.WebContents & {
+	kernelWindowData?: {
+		originalPreload: string;
+		windowOptions: electron.BrowserViewConstructorOptions;
+	}
+}
+
 export const patches: {
 	[id: string]: PatchFunction;
 } = {};
 
 export const originalBrowserWindow = electron.BrowserWindow;
 
+let BrowserWindowProxy = null;
 const preloadPath = path.join(__dirname, "..", "..", "preload");
 
 patchElectron("BrowserWindow", (target, property) => {
 	if (property === "BrowserWindow") {
 		// Create a Proxy over Electron that returns the PatchedBrowserWindow if BrowserWindow is called.
 		// Can't just proxy the BrowserWindow class directly because it's a getter.
-		return new Proxy(target.BrowserWindow, {
+		BrowserWindowProxy ??= new Proxy(target.BrowserWindow, {

```

Comment:
Will this impact the creation of multiple BrowserWindows?

Reply:
No, it's just unnecessary to re-create the proxy every time something accesses it.